File: docs-examples/example-java/src/main/java/io/micronaut/configuration/openapi/docs/generics/NamedResponseController.java
package io.micronaut.configuration.openapi.docs.generics;

import io.micronaut.configuration.openapi.docs.schemas.Pet;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Put;
import io.swagger.v3.oas.annotations.media.Schema;

@Controller("/named")
class NamedResponseController {

    // tag::method[]
    @Put("/")
    @Schema(name = "ResponseOfPet")
    public Response<Pet> updatePet(Pet pet) {
        Response<Pet> response = new Response<>();
        response.setResult(pet);
        return response;
    }
    // end::method[]
}

File: docs-examples/example-kotlin/src/main/kotlin/io/micronaut/configuration/openapi/docs/generics/NamedResponseController.kt
package io.micronaut.configuration.openapi.docs.generics

import io.micronaut.configuration.openapi.docs.schemas.Pet
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Put
import io.swagger.v3.oas.annotations.media.Schema

@Controller("/named")
class NamedResponseController {

    // tag::method[]
    @Put("/")
    @Schema(name = "ResponseOfPet")
    fun updatePet(pet: Pet): Response<Pet> {
        return Response(pet)
    }
    // end::method[]
}
